DIPs/1NNN-DK.md

```diff
@@ -0,0 +1,242 @@
+
+## Contents
+* [Rationale](#rationale)
+* [Prior Work](#prior-work)
+* [Description](#description)
+* [Breaking Changes and Deprecations](#breaking-changes-and-deprecations)
+* [Reference](#reference)
+* [Copyright & License](#copyright--license)
+* [Reviews](#reviews)
+
+## Rationale
+
+D supports [many attributes](https://dlang.org/spec/attribute.html).
+When a multitude of attributes needs to apply to a multitude of declarations, it can lead to visual noise from
+repeating the attribute names, colloquially called 'attribute soup'.
+
+```D
+struct S
+{
+    private void f() const @nogc nothrow pure @safe
+    {
+
+    }
+
+    private void g() const @nogc nothrow pure @safe
+    {
+
+    }
+
+    // ...
+}
+```
+
+One way to mitigate this is by using the `Attribute:` syntax, which will make the attribute apply to all following declarations until the end of the scope the attribute appears in.
+
+```D
+struct S
+{
+    @nogc nothrow pure @safe:
+    private:
+    const:
+
+    void f()
+    {
+
+    }
+
+    void g()
+    {
+
+    }
+
+    // ...
+}
+// attributes' influence ends here
+```
+
+This reduces the repetition of attributes, but it becomes a problem when there are a few declarations that need an exception.
+Sometimes this can be solved by annotating the exceptional declaration with an inverse attribute:
+
+| Attribute                | Undone by           |
+|--------------------------|---------------------|
+| `private`                | `public`            |
+| `@system`                | `@safe`             |
+| `extern(C)`              | `extern(D)`         |
+| `nothrow`                | `throw`             |
+| `pragma(inline, true)`   | `pragma(inline)`    |
+
+However, most attributes do not have an inverse that resets it to the default:
+
+- `align()`
+- `deprecated`
+- `static`
+- `extern`
+- `abstract`
+- `final`
+- `override`
+- `synchronized`
+- `scope`
+- `const`
+- `immutable`
+- `inout`
+- `shared`
+- `__gshared`
+- `pure`
+- `@disable`
+- `@nogc`
+- `@property`
+- `@UserDefinedAttribute`
+
+Moreover, because function attributes are inferred in template functions, forcing them back to the default is not always desirable either.
+Consider:
+
+```
+struct S
+{
+    @nogc nothrow pure @safe:
+
+    void f() { }
+
+    void g() { }
+
+    void toString(O)(ref O sink) if (isOutputRange!(O, char))
+    {
+    }
+}
+```
+
+In this case, the `toString` method should not force the provided Output Range to have an implementation of `sink.put` with all function attributes, but
+neither should `toString` be denied those attributes when it can infer them.
+The only option here is to place all template methods above the `@nogc nothrow pure @safe:` line, or go back to 'attribute soup' by repeating all function attributes for individual members.
+
+What is really desired is something to specify that attributes on `toString` should go back to the default state,
+which this DIP proposes in the form of the `@default` attribute.
```

Comment:
Especially for the case of templates, alternatively, we could change the language so that attributes are inferred unless they are applied directly to the template. Colon and brace attributes need not matter. With the proposal, `@default` is a niche thing I have to remember.